### PR TITLE
Added previous delegate calls for UINavigationController

### PR DESCRIPTION
**Pull request by bradphilips**
This allows the calling application to assign a delegate to the `UINavigationController` to handle `navigationController:didShow` and `navigationController:willShow` delegate events.

### CODE REVIEW

File: Podfile.lock [hunk 1]
```diff
@@ -23,4 +23,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: f1dbc884f10a7b2ddff5039dfc73e33610decc4b
 
-COCOAPODS: 1.3.1
+COCOAPODS: 1.4.0
```
Comment (lkzhao) [inline Podfile.lock]: Lets not update cocoapods version in this PR

**Comment by rickbdotcom** [comment]
Would be nice to have this for UITabBarDelegate as well.

File: Podfile.lock (hunk 1 above)
Comment (bradphilips) [reply Podfile.lock]: This was unintended, apologies. Reverted it, thanks.

**Comment by bradphilips** [comment]
@lkzhao, I've reverted the change for the Podfile. This was a mistake. Thanks!

**Comment by lkzhao** [comment]
Awesome Thanks!

**Pull request merged**